core_impl/src/main/java/io/opencensus/implcore/stats/StatsComponentImplBase.java

```diff
@@ -55,11 +55,15 @@ public StatsRecorderImpl getStatsRecorder() {
 
   @Override
   public StatsCollectionState getState() {
-    return state;
+    state.setRead();
+    return state.get();
   }
 
   @Override
   public void setState(StatsCollectionState newState) {
-    state = Preconditions.checkNotNull(newState, "newState");
+    state.set(Preconditions.checkNotNull(newState, "newState"));
+    if (newState == StatsCollectionState.DISABLED) {
+      viewManager.clearStats();
+    }
```

Comment:
I think this method also needs to reset the cumulative view start times when stats collection is changed from disabled to enabled.